### ISSUE TITLE
Question about the results

### ISSUE THREAD

**Issue by Alex-Wula** [md4all/md4all#18]
Dear authors:
Thanks for your work! These days, I try to run the code you provided in this repository. I used the baseline_nuscenes.ckpt and md2_nuscnes.ckpt as the teacher net to train the model on the NuScenes-v1.0 dataset respectively. However, the results is different from the results you offered in table 1. The results of baseline_nuscenes.ckpt deviate somewhat from table1, while the results of md2_nuscenes.ckpt are very poor if I don't align them with ground truth. When align the results of md2_nuscenes.ckpt, the results still worse than the results in table1. 
I also use the evaluate_depth.py to evaluate the trained model you offered called md4allDDa_nuscenes.ckpt. The results of this model is close to the data in Table 1, but still worse than the data in the table.
My environments are as followed: Python 3.8.18, Pytorch 2.1.0, torchvision 0.16.0, numpy 1.24.2, Pillow 10.1, and all the experiments  carried on a single RTX 4090.
I wonder if I chose the wrong teacher net? Or maybe the environment leads to a worse results?
I have put the results into the table, please check！Thank you very much!
baseline_nuscenes:
![baseline_nuscenes](https://github.com/md4all/md4all/assets/154568622/848dc078-e811-4a93-aa4b-4b3f3175b705)
md2_nuscenes:
![md2_nuscenes](https://github.com/md4all/md4all/assets/154568622/900eb718-38d8-4582-885d-dc3238997cdf)
md4allDDa_nuscenes:
![md4allDDa_nuscenes](https://github.com/md4all/md4all/assets/154568622/38e91ce8-fa69-44e3-8a2b-327af23f7efc)

**Comment by sgasperini** [comment]
Dear @Alex-Wula,

Thank you for your interest in our work! Sorry for the delay, it's the holiday season here.

If I understood correctly, the issues raised are all about evaluating the checkpoints we provided and are not about trainings of your own. I am not sure if this assessment is correct since, in the beginning, you wrote about training something yourself, but later in your message, you addressed our pretrained models as checkpoints. Please clarify if this is not the case.

You wrote that md2 was used to train the model, but that is not true. We first trained ForkGAN (we already provide the translated images, so no need to train ForkGAN again), then a baseline model (md2+velocity supervision, we provide this too), and finally, our md4all model. This multi-stage procedure is also described in issue #6, along with the commands to achieve that.

For your md4allDDa_nuscenes evaluation, the results are nearly identical to the ones we reported in Table 1, the difference is really minor at the 4th decimals (pp values). Such minor differences are likely due to the different environments, as you hypothesized.
For the other models, the differences are indeed significant. Please clarify whether you used the provided Docker instructions and whether you followed the instructions in the readme on evaluating the models. It is quite bizarre since you could reproduce our evaluation of the md4allDDa model on nuScenes we provided but did not reach the same values as Table 1 when evaluating the baseline and md2 models we provided [here](https://drive.google.com/drive/folders/1nylOZitf8P33vlq-TKEcGte0FCQQ1wR5?usp=sharing). Or were those md2 and baseline models something you trained from scratch?

Regarding the _gt results, please consider that this topic has already been discussed in other issues (#8 and #11). Briefly, md2 needs the gt scaling as it is not scale-aware, so it is normal that the results without _gt are not good for md2. Instead, the baseline and ours are scale-aware (thanks to the velocity supervision from PackNet), so the reference values are without _gt (i.e., the pp ones).

Please look at the linked issues and the commands reported in #6 and the readme. What are the differences between what you ran and what we reported and wrote? We will need more details from you to further support you.

**Comment by sgasperini** [comment]
Following up on the previous comment, as we wrote in #6, the baseline checkpoint we provided on Google Drive is the one we used to train md4all, but it is different from the one we reported in the table. The one from the table was coming from an older codebase.
This explains the discrepancy for the baseline, so what remains unclear are the md2 results.

**Comment by Alex-Wula** [comment]
Dear @sgasperini
Thank you very much for your reply! These days, I read the code and the readme again, and I also read your reply yesterday. In fact, I used the checkpoints(baseline_nuscenes.ckpt and md2_nuscenes.ckpt) as the teacher models to train the md4all-DD, and the results are shown in the pic.1 and pic.2 .  After reading your reply, I knew that I should use the baseline_nuscenes.ckpt as the teacher net. 
The results of pic.1 come from the md4all-DD model I used the baseline_nuscenes.ckpt as its teacher net.  I will retrain the model and if I have any questions, I will continue to reply to you below this issue.
Thank you very much for your help, I really appreciate it !

**Comment by Alex-Wula** [comment]
Dear @sgasperini 
I recently re-ran the experiment and successfully reproduced your results. Although the numerical value is lower than the result of md4all-DD in your Table 1, I think this result is within the acceptable range. Thank you very much for your reply! Your work helps me a lot! ![2F9C3F19](https://github.com/md4all/md4all/assets/154568622/3d7124ed-8af7-460c-9369-da4e088308e6)

**Comment by sgasperini** [comment]
Great to hear @Alex-Wula!! Thanks a lot for updating us and the feedback!

As a rule of thumb, if the values differ at the 4th decimal, then it's likely the environment. Small variations at the third decimal could be due to the randomness during training (even though we use a seed, some operations are inherently random). Differences at the second decimal suggest there's a bug (at least for values being similar to 0.1234).
Good luck with your research!

**Comment by thaitran24** [comment]
Dear @Alex-Wula, can you provide me your reproduced evaluation results of the md4all-DDa model on nuScenes? Thank you very much!

**Comment by Alex-Wula** [comment]
Dear @thaitran24 
Sure! But I'm sorry that I'm quite busy this week. I'll try to provide you with the results as soon as possible. Do you want me to send it to you via email or put it directly under this issue?

**Comment by thaitran24** [comment]
Thank you very much! Take your time and you can send it under this issue when you're free 😁

**Comment by Alex-Wula** [comment]
@thaitran24 Here is the reproduced results. The first row is the results provided in the paper. The second row is the results of the checkpoint author provided. The third row is the results I reproduced. Experiments carried out on a single 4090.  
![image](https://github.com/md4all/md4all/assets/154568622/4c298e17-e69c-4072-a932-774865e502fb)
